service/src/teeracle/mod.rs

```diff
@@ -27,99 +28,81 @@ use substrate_api_client::{SubmitAndWatch, XtStatus};
 use teeracle_metrics::{increment_number_of_request_failures, set_extrinsics_inclusion_success};
 use tokio::runtime::Handle;
 
-pub(crate) mod interval_scheduling;
+pub(crate) mod schedule_periodic;
 pub(crate) mod teeracle_metrics;
 
-/// Send extrinsic to chain according to the market data update interval in the settings
-/// with the current market data (for now only exchange rate).
-pub(crate) fn start_interval_market_update<E: TeeracleApi>(
-	api: &ParentchainApi,
+pub(crate) fn schedule_teeracle_reregistration_thread(
+	send_register_xt: impl Fn() -> Option<Hash> + std::marker::Send + 'static,
 	interval: Duration,
+) {
+	println!("Schedule the teeracle reregistration every: {:?}", interval);
+
+	std::thread::Builder::new()
+		.name("teeracle_reregistration_thread".to_owned())
+		.spawn(move || {
+			schedule_periodic(
+				|| {
+					println!("Reregistering the teeracle.");
```

Comment:
I think this could be `trace` or `debug` here.

Reply:
I have been wondering about this, too. But I thought, as this is part of the regular operation in production, I think we want to see this always. Also this happens roughly only every 24 hours, so I think this will not spam the logs.

Reply:
Ahhh, sorry. I totally agree. This one should be trace/debug, and only the one two lines after should be a print statement.